### PR TITLE
Add `BernoulliLogit`

### PR DESCRIPTION
**Pull request by devmotion**
This PR adds a Bernoulli distribution with the logit of the success rate as parameter.

This is motivated by use cases e.g. in Turing (https://github.com/TuringLang/Turing.jl/issues/1890) where it is more convenient to work with unconstrained parameters. First class support for the parameterization allows us to evaluate e.g. the log pdf more efficiently.

### CODE REVIEW

**Comment by mschauer** [comment]
Can you mention “parametrized by the log of the odds `Logitp = p/(1-p)`” in the docstring

**Comment by devmotion** [comment]
> Can you mention “parametrized by the log of the odds `Logitp = p/(1-p)`” in the docstring

Are you interested in the term "log-odds" specifically? The docstring already mentions that it is parameterized by the logit of the success rate, and log-odds is just another name for the logit (see e.g. https://en.wikipedia.org/wiki/Logit).

**Comment by devmotion** [comment]
Generally this PR should be ready and well-tested.

File: test/univariate/discrete/bernoullilogit.jl [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            @test @inferred(cquantile(d, q)) ≈ cquantile(d0, q) nans=true
+        end
+
+        for t in (-5.2, 1.2f0)
+            @test @inferred(mgf(d, t)) ≈ mgf(d0, t) rtol=1e-6
+            @test @inferred(cgf(d, t)) ≈ cgf(d0, t) rtol=1e-6
+            @test @inferred(cf(d, t)) ≈ cf(d0, t) rtol=1e-6
+        end
+    end
+end
```
Comment (matbesancon) [inline test/univariate/discrete/bernoullilogit.jl:81]:
```suggestion
end

```

**Review by matbesancon — APPROVED** [review]

**Comment by matbesancon** [comment]
should be good to merge when tests pass I think

**Comment by mschauer** [comment]
> The docstring already mentions that it is parameterized by the logit of the success rate

Well, that requires you knowing what a logit is. Saying it is "parametrised by `logitp = p/(1-p)` where `p` is the success rate" makes everything immediately clear to the passerby - doesn't it?

File: src/univariate/discrete/bernoullilogit.jl [hunk 2]
```diff
@@ -0,0 +1,108 @@
+"""
+    BernoulliLogit(logitp=0.0)
+
+A *Bernoulli distribution* that is parameterized by the logit `logitp` of its success rate.
```
Comment (devmotion) [inline src/univariate/discrete/bernoullilogit.jl:4]:
```suggestion
A *Bernoulli distribution* that is parameterized by the logit `logitp = logit(p) = log(p/(1-p))` of its success rate `p`.
```

**Comment by devmotion** [comment]
Ah I see, I thought the main point of your comment was the term log odds. I changed the docstring, hopefully it is clearer now.

**Review by mschauer — APPROVED** [review]

**Pull request merged**